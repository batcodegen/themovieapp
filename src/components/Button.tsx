import {StyleSheet, Text, TouchableOpacity} from 'react-native';
import React, {memo} from 'react';

interface ButtonProps {
  title: string;
  disabled?: boolean;
  onButtonPress: () => void;
}

const Button = ({title, disabled = false, onButtonPress}: ButtonProps) => {
  const bgButtonColor: HEX = disabled ? '#868583' : '#f0af0c';
  return (
    <TouchableOpacity
      style={[styles.container, {backgroundColor: bgButtonColor}]}
      disabled={disabled}
      onPress={onButtonPress}>
      <Text style={styles.text}>{title}</Text>
    </TouchableOpacity>
  );
};

export default memo(Button);

const styles = StyleSheet.create({
  container: {
    margin: 20,
    width: '70%',
    padding: 10,
    alignItems: 'center',
    borderRadius: 10,
  },
  text: {color: 'white', fontSize: 16, fontWeight: 'bold'},
});
