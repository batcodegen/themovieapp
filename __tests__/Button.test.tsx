import React from 'react';
import {render, screen} from '@testing-library/react-native';
import Button from '../src/components/Button';

describe('Login', () => {
  test('renders Login correctly', async () => {
    const mockButtonPress = jest.fn();
    render(
      <Button onButtonPress={mockButtonPress} title="abc" disabled={false} />,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
