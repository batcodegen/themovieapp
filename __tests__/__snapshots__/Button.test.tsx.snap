// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders Login correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f0af0c",
      "borderRadius": 10,
      "margin": 20,
      "opacity": 1,
      "padding": 10,
      "width": "70%",
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    abc
  </Text>
</View>
`;
